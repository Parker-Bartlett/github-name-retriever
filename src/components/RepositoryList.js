import React, { Component } from 'react';
import Repository from './Repository'

class Search extends Component {
  
  render() {
    return (
      <div style={repoList}>
        <h3 data-testid="header">Public Repositories Retrieved: {this.props.repos.length}</h3>
        <ul>
          {this.props.repos.map(repo => (
            <Repository key={repo.id} repo={repo} />
          ))}
        </ul>
      </div>
    )
  }
}

const repoList = {
  margin: '10px auto',
  padding: '10px',
  width: '90%',
  textAlign: 'center',
  backgroundColor: 'rgba(0, 0, 0, .1)',
  border: '1px solid black'
} 

export default Search;
