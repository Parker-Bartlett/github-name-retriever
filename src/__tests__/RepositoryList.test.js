import React from 'react';
import { render, cleanup } from 'react-testing-library';
import RepositoryList from '../components/RepositoryList'

afterEach(cleanup)

const repos = [
    {id: 1},
    {id: 2},
    {id: 3},
]
test('<RepositoryList> With Repos', () => {
    const { getByTestId, getAllByTestId, container } = render(<RepositoryList repos={repos} />)
    
    expect(getByTestId('header').innerHTML).toBe(`Public Repositories Retrieved: ${repos.length}`)
    expect(getAllByTestId('list-item').length).toBe(3);

    expect(container.firstChild).toMatchSnapshot();
})