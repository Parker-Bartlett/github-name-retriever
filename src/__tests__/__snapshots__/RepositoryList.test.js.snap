// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepositoryList> With Repos 1`] = `
<div
  style="margin: 10px auto; padding: 10px; width: 90%; text-align: center; background-color: rgba(0, 0, 0, 0.1); border: 1px solid black;"
>
  <h3
    data-testid="header"
  >
    Public Repositories Retrieved: 
    3
  </h3>
  <ul>
    <div>
      <li
        data-testid="list-item"
        style="list-style-type: none; padding: 5px; display: flex; justify-content: space-between;"
      >
        <a
          data-testid="repo-link"
          target="blank"
        />
        <p
          data-testid="watchers"
        >
          Watchers: 
        </p>
      </li>
    </div>
    <div>
      <li
        data-testid="list-item"
        style="list-style-type: none; padding: 5px; display: flex; justify-content: space-between;"
      >
        <a
          data-testid="repo-link"
          target="blank"
        />
        <p
          data-testid="watchers"
        >
          Watchers: 
        </p>
      </li>
    </div>
    <div>
      <li
        data-testid="list-item"
        style="list-style-type: none; padding: 5px; display: flex; justify-content: space-between;"
      >
        <a
          data-testid="repo-link"
          target="blank"
        />
        <p
          data-testid="watchers"
        >
          Watchers: 
        </p>
      </li>
    </div>
  </ul>
</div>
`;
